other-docs/guides/headless/README.md

```diff
@@ -0,0 +1,56 @@
+
+
+### Semi-decoupled
+
+A semi-decoupled architecture is a hybrid architecture incorporating parts of both coupled and decoupled architectures. It allows many of the benefits of decoupled architectures to be realised while avoiding much of the overhead.
+
+With a semi-decoupled system, the frontend is built as a part of the application, but has only a few, well-defined integration points between the frontend and backend. These are kept small and manageable to allow teams to work mostly independently. The frontend is built in JavaScript, allowing in-browser rendering in combination with [server-side rendering](ssr.md). Both the frontend and backend run on the same infrastructure.
+
+Platform provides a REST API to allow building semi-decoupled frontends, in addition to tools and libraries to facilitate building these frontends. The [Building a React App guide](react-app.md) guide walks through building a semi-decoupled React app from scratch.
+
+Additionally, Platform provides infrastructure to allow for [server-side rendering](ssr.md).
+
+
+## Deciding on architecture
+
+Before embarking on a headless architecture, decide exactly how you want to build and manage your project.
+
+Headless frontends can be beneficial in allowing separate teams and skills to handle building a site's frontend. However, they also represent a significant amount of work, as it often requires rebuilding significant pieces of functionality to better fit the decoupled architecture. For example, routing requests is a mostly-solved problem with traditional coupled architectures, but can require recreating the routing system in a decoupled architecture.
+
+Additionally, this may require separate and complex infrastructure for serving web requests. This can increase running and maintenance costs, as multiple systems need to be run.
+
+Decoupling can also have overhead as compared to monolithic systems, as data needs to be passed between multiple systems. This can cause both technical overhead with latency on round-trips, as well as mental overhead during development, as systems need to be specifically built to accomodate this. Consider whether the additional development time is worth the cost of decoupling.
+
+In many cases, a semi-decoupled architecture can achieve the best of both worlds. This uses parts of the theming system for some frontend functionality, while moving much of the work off to a separate frontend. This can help avoid duplication across systems while still providing many of the benefits of a decoupled architecture. In combination with server-side rendering, data can be generated at the same time as rendering the frontend for users, providing an fluid interactive experience to users with little overhead.
+
+Generally, we recommend using a semi-decoupled architecture, but each case is different. The HM Platform team can provide consultancy and support to guide you through this process.
```

Comment:
I think we could document some more of the things you lose when going de-coupled, thinking: Customizer, previews, logged in behaviours on the front-end, admin bar,